### PR TITLE
init quiescence after all templates are rendered

### LINKED ISSUE #1229: With "wait" consul-template spawns the child process before rendering the template
### Consul Template version
`consul-template v0.20.0`
(The same setup works fine in `consul-template v0.19.5`)

### Configuration

/etc/consul-template.conf:

```hcl
consul {
  address = "consul.query.consul:28500"
  retry {
    enabled = true
    attempts = 12
    backoff = "250ms"
    max_backoff = "1m"
  }
}
reload_signal = "SIGHUP"
kill_signal = "SIGINT"
log_level = "warn"
pid_file = "/var/run/consul-template/consul-template.pid"
exec {
  command = "/usr/local/bin/pgpool -n -D"
  kill_signal = "SIGINT"
  kill_timeout = "10s"
}
template {
  source = "/usr/local/etc/pgpool.conf.tpl"
  destination = "/usr/local/etc/pgpool.conf"
  wait {
    min = "5s"
    max = "10s"
  }
}
```

### Command

```shell
consul-template -config /etc/consul-template.conf
```

### Expected behavior

After the wait time, it should render the template and then spawn the child process.

### Actual behavior

It spawns the child process before rendering the template (only when you use `wait` in the `template` block). It seems that it executes the command in the `exec` block and then waits for rendering the template.

### PR DESCRIPTION
**Pull request by eikenb**
Hold off on initializing wait/quiescence until all templates have been
rendered once. This fixes an issue with -exec firing before the template
has been rendered and would also make sure templates are rendered for
once (if that wasn't already fixed another way).

Fixes #1229

### CODE REVIEW

**Comment by eikenb** [comment]
There is no test case for it currently. I plan on fixing that before merging this PR. Just wanted to get it out in front of people sooner.

**Comment by eikenb** [comment]
@pearkes .. I added the test to reproduce it.

File: manager/runner_test.go [hunk 1]
```diff
@@ -780,6 +780,74 @@ func TestRunner_Start(t *testing.T) {
...
+			Consul: &config.ConsulConfig{
+				Address: config.String(testConsul.HTTPAddr),
+			},
+			Wait: &config.WaitConfig{
+				Min: config.TimeDuration(5 * time.Millisecond),
+				Max: config.TimeDuration(10 * time.Millisecond),
+			},
+			Exec: &config.ExecConfig{
+				// `cat filename` would fail if template hadn't rendered
+				Command: config.String(`cat ` + firstOut.Name()),
```
Comment (pearkes) [inline manager/runner_test.go]: Neat 😄. Took me a second to realize that this is the vital part of the test but totally makes sense. Are there any cross-platform concerns here? Not sure we can really get around it given we're testing `exec`...

Comment (eikenb) [reply manager/runner_test.go]: The `cat` command is POSIX standard which is about as much as I think is possible for cross-platform support.

Comment (pearkes) [reply manager/runner_test.go]: Yeah makes sense.

**Pull request merged**